### DISCUSSION TITLE
[QUESTION] How can I turn off RTE for specific block types?

### DISCUSSION THREAD

**Discussion by LordRembo** [GrapesJS/grapesjs#6179] [Q&A]
The issue I'm facing, is that I'm dealing with an implementation of GrapesJS where we don't want users to style text at all. 
We've just got plain text we want to edit, as part of a button (`<a>` tag).
On top of that, we're using CKE5 which wraps text in `p` tags and that's not something you can turn off (CKE5 always needs text to be wrapped in block-level tags, and they're very adamant about not changing that).
So the easiest way to get rid of both of these issues at once, would be to just have a plain textarea instead of allowing an RTE to render.
Any ideas how we could accomplish this?

**Comment by artf** [comment]
You should be able to update/extend the text component view (eg. `onActive` method) and apply there your logic of when to proceed or not with the rich text editor.

**Comment by LordRembo** [comment]
@artf The problem is, I can't find anything in the docs abut disabling the RichTextEditor specifically for a component. [I did find info about the view of a component type](https://grapesjs.com/docs/modules/Components.html#define-custom-component-type). So using view -> onActive  would contain the logic. But I don't see a way to make it load a textarea without the RTE being active.

**Reply by artf** [reply]
Yeah unfortunately it's hard to document every custom use case, so I'd suggest checking the [source of the method](https://github.com/GrapesJS/grapesjs/blob/d74c3f18a873bac5bd9658659932acf165c91ef5/packages/core/src/dom_components/view/ComponentTextView.ts#L74) and figure out how to setup your own logic.

**Reply by kafeltz** [reply]
Grave news for anyone trying to disable the unwanted bar. For a specific type or custom-type.

It is impossible to disable the RTE, even when copying and pasting the original source code and overwriting it, trying not loading it. If the RTE toolbar is not shown, the editor is uneditable.

Even ```rte.disable()``` does not work.

**Reply by LordRembo** [reply]
@kafeltz Can you share the code for how you tried it? If it can prevent the RTE toolbar, we must be able to also reach the textarea and enable that.

**Reply by LordRembo** [reply]
How I'm approaching it now, is use onActive event to build the complete content of the modal using `view` to render (for example) an input field to enter content in, and a submit button with custom login that updates the email content using `editor.getSelected()`.set('content', …`
